xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -1301,60 +1305,85 @@ bool CDVDInputStreamNavigator::IsSubtitleStreamEnabled()
   return m_dll.dvdnav_get_active_spu_stream(m_dvdnav) >= 0;
 }
 
-bool CDVDInputStreamNavigator::GetState(std::string &xmlstate)
+bool CDVDInputStreamNavigator::FillDVDState(DVDState& dvdState)
 {
-  if( !m_dvdnav )
+  if (!m_dvdnav)
+  {
     return false;
+  }
 
-  dvd_state_t save_state;
-  if( DVDNAV_STATUS_ERR == m_dll.dvdnav_get_state(m_dvdnav, &save_state) )
+  if (m_dll.dvdnav_current_title_program(m_dvdnav, &dvdState.title, &dvdState.pgcn,
+                                         &dvdState.pgn) == DVDNAV_STATUS_ERR)
   {
-    CLog::Log(LOGWARNING, "CDVDInputStreamNavigator::GetNavigatorState - Failed to get state ({})",
-              m_dll.dvdnav_err_to_string(m_dvdnav));
+    CLog::LogF(LOGERROR, "Failed to get current title info ({})",
+               m_dll.dvdnav_err_to_string(m_dvdnav));
     return false;
   }
 
-  if( !CDVDStateSerializer::DVDToXMLState(xmlstate, &save_state) )
+  int current_angle = GetActiveAngle();
+  if (current_angle == -1)
   {
-    CLog::Log(LOGWARNING, "CDVDInputStreamNavigator::SetNavigatorState - Failed to serialize state");
+    CLog::LogF(LOGERROR, "Could not detect current angle, ignoring saved state");
     return false;
   }
+  dvdState.current_angle = current_angle;
+
+  dvdState.time = m_dll.dvdnav_get_current_time(m_dvdnav) / 90;
+  dvdState.audio_num = GetActiveAudioStream();
+  dvdState.subp_num = GetActiveSubtitleStream();
+  dvdState.sub_enabled = IsSubtitleStreamEnabled();
 
   return true;
 }
 
-bool CDVDInputStreamNavigator::SetState(const std::string &xmlstate)
+bool CDVDInputStreamNavigator::GetState(std::string& xmlstate)
```

Comment:
Are you sure you want `std::string&`? Usually function parameters don't allow side effects.

Reply:
think it needs to be non-const since we‘re getting the state and store it into xmlstate?

Reply:
Yes, I read this wrong. Disregard!!